libraries/AP_InertialSensor/AP_InertialSensor_RST.cpp

```diff
@@ -0,0 +1,436 @@
+        return false;
+    }
+
+    // set flag for reading registers
+    _dev_gyro->set_read_flag(0x80);
+
+    _dev_gyro->set_speed(AP_HAL::Device::SPEED_HIGH);
+
+    _dev_gyro->read_registers(GYRO_WHO_AM_I, &_gyro_whoami, sizeof(_gyro_whoami));
+    if (_gyro_whoami != GYRO_WHO_I_AM) {
+        hal.console->printf("RST: unexpected gyro WHOAMI 0x%x\n", (unsigned)_gyro_whoami);
+        printf("RST: unexpected gyro WHOAMI 0x%x\n", (unsigned)_gyro_whoami);
+        goto fail_whoami;
+    }
+
+    printf("detect i3g4250d\n");
+
+    //enter power-down mode first
+    _dev_gyro->write_register(GYRO_CTRL_REG1, 0);
+
+    hal.scheduler->delay(100);
+
+    _dev_gyro->write_register(GYRO_CTRL_REG1, 
+                              GYRO_REG1_POWER_NORMAL | GYRO_REG1_Z_ENABLE | GYRO_REG1_X_ENABLE | GYRO_REG1_Y_ENABLE | RATE_800HZ_LP_50HZ);
+
+	_dev_gyro->write_register(GYRO_CTRL_REG2, 0);		/* disable high-pass filters */
+	_dev_gyro->write_register(GYRO_CTRL_REG3, 0x0);        /* DRDY disable */
+	_dev_gyro->write_register(GYRO_CTRL_REG4, RANGE_2000DPS);
+	_dev_gyro->write_register(GYRO_CTRL_REG5, GYRO_REG5_FIFO_ENABLE);		/* disable wake-on-interrupt */
+
+	/* disable FIFO. This makes things simpler and ensures we
+	 * aren't getting stale data. It means we must run the hrt
+	 * callback fast enough to not miss data. */
+	_dev_gyro->write_register(GYRO_FIFO_CTRL_REG, GYRO_FIFO_CTRL_BYPASS_MODE);
+
+	_gyro_scale = 70e-3f / 180.0f * M_PI;
+
+    hal.scheduler->delay(100);
+
+    _gyro_spi_sem->give();
+
+    return true;
+
+fail_whoami:
+    _gyro_spi_sem->give();
+    return false;
+
+
+}
+
+/*
+ * init accel
+ */
+bool AP_InertialSensor_RST::_init_accel(void)
+{
+    _accel_spi_sem = _dev_accel->get_semaphore();
+
+    if (!_accel_spi_sem->take(HAL_SEMAPHORE_BLOCK_FOREVER)) {
+        return false;
+    }
+
+    _dev_accel->set_speed(AP_HAL::Device::SPEED_HIGH);
+
+    _dev_accel->set_read_flag(0x80);
+
+    _dev_accel->read_registers(ACCEL_WHO_AM_I, &_accel_whoami, sizeof(_accel_whoami));
+    if (_accel_whoami != ACCEL_WHO_I_AM) {
+        hal.console->printf("RST: unexpected accel WHOAMI 0x%x\n", (unsigned)_accel_whoami);
+        printf("RST: unexpected accel WHOAMI 0x%x\n", (unsigned)_accel_whoami);
+        goto fail_whoami;
+    }
+
+	_dev_accel->write_register(ACCEL_CTRL_REG1,
+                               ACCEL_REG1_POWER_NORMAL | ACCEL_REG1_Z_ENABLE | ACCEL_REG1_Y_ENABLE | ACCEL_REG1_X_ENABLE | RATE_1000HZ_LP_780HZ);
+	_dev_accel->write_register(ACCEL_CTRL_REG2, 0);		/* disable high-pass filters */
+	_dev_accel->write_register(ACCEL_CTRL_REG3, 0x02);        /* DRDY enable */
+	_dev_accel->write_register(ACCEL_CTRL_REG4, ACCEL_REG4_BDU | ACCEL_REG4_FULL_SCALE_8G);
+
+    _accel_scale = 0.244e-3f * ACCEL_ONE_G;
+
+    _accel_spi_sem->give();
+
+    return true;
+
+fail_whoami:
+    _accel_spi_sem->give();
+    return false;
+
+
+}
+
+bool AP_InertialSensor_RST::_init_sensor(void)
+{
+    if(!_init_gyro() || !_init_accel())
+    {
```

Comment:
Style guide mandates a space after the *if* and that this brace goes in the same line as the *if*. There are more cases below.